README.md

```diff
@@ -1 +1,53 @@
-# istio-csr
\ No newline at end of file
+# istio-csr
+
+cert-manager-istio-agent is an agent which allows for [istio](https://istio.io) workload
+and control plane components to be secured using
+[cert-manager](https://cert-manager.io). Certificates facilitating mTLS, inter
+and intra cluster, will be signed, delivered and renewed using [cert-manager
+issuers](https://cert-manager.io/docs/concepts/issuer).
+
+Currently supports istio v1.7
+
+---
+
+## Installation
+
+Firstly, [cert-manager must be
+installed](https://cert-manager.io/docs/installation/) in your cluster. An
+issuer must be configured, which will be used to sign your certificate
```

Comment:
Are there any important constraints on the issuer? e.g. does it have to be a clusterissuer?

Reply:
No, you can have whatever you want:

```
  -g, --issuer-group string             Group of the issuer to sign istio workload certificates. (default "cert-manager.io")
  -k, --issuer-kind string              Kind of the issuer to sign istio workload certificates. (default "Issuer")
  -u, --issuer-name string              Name of the issuer to sign istio workload certificates. (default "istio-ca")
```

Reply:
If it's an Issuer presumably it has to be in the namespace where you are running the agent? That's probably obvious?

Reply:
No, in the default installation, the agent runs in cert-manager, CRs and Issuer are in istio-system.

You can configure the target namespace with;

```bash
  -c, --certificate-namespace string    Namespace to request certificates. (default "istio-system")
```

I would have thought the Issuer being in the istio-system being obvious but perhaps not..

Reply:
A good rule of thumb is that nothing is as obvious as you think :)

Reply:
A rule to live by.. I'll update the README.md